web-ide/proxy/src/routes/httpToHttps.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Application } from "express"
+const conf = require('../config').read()
+
+/**
+ * Due to limitations in our gcp load balancers we implement redirects to https ourselves, on a separate port
+ */
+export default class HttpToHttpsRoute {
+    redirectApp :Application
+    constructor(redirectApp :Application) {
+        this.redirectApp = redirectApp
+    }
+
+    init() : HttpToHttpsRoute {
+        this.redirectApp.get('*', (req,res) => {
+            if(!req.secure) {
+                const r = ['https://', req.hostname, ":", 443, req.url].join('')
+                return res.redirect(307, r);
```

Comment:
Why `307` rather than `301`, i.e., why a temporary redirect instead of a permanent redirect?

Reply:
Because we might change our mind in future and don't want browsers to go straight to the new location. A permanent redirect is often a bad idea.

Reply:
Unconvinced we won't go behind an Nginx style redirector, in which case we'd then ditch having https on this server at al.

Reply:
I noticed odd behavior in chrome without the temporary. It started redirecting things I didn't want, like other ports on the same host!